### ISSUE TITLE
Scene can't have std::vector member

### ISSUE THREAD

**Issue by redminion** [cocos2d/cocos2d-x#12962]
Creating a class that inherits from cocos2d::Scene, and giving it a std::vector member variable causes a runtime error but compiles fine.  The error I get is "vector iterators incompatible" and occurs on line457 of CCNavMesh.cpp.  (Im running windows7 and visual studio 2013 update 4)

This bug is related to the physics code since disabling CC_USE_PHYSICS no longer produces an error.

This bug is new to v3.7.

**Comment by ricardoquesada** [comment]
can you post the code that reproduces the bug? thanks.

**Comment by redminion** [comment]
To reproduce the problem create a brand new c++ project.

(1) --- Edit HelloWorldScene.h to inherit Scene and have member vector:

```
#ifndef __HELLOWORLD_SCENE_H__
#define __HELLOWORLD_SCENE_H__

#include "cocos2d.h"

class HelloWorld : public cocos2d::Scene
{
public:
    std::vector<int> myMemberVector;

    static cocos2d::Scene* createScene();
    virtual bool init();
    void menuCloseCallback(cocos2d::Ref* pSender);
    CREATE_FUNC(HelloWorld);
};

#endif // __HELLOWORLD_SCENE_H__
```

(2) --- Edit HelloWorldScene.cpp to make createScene() return a Scene:

```
Scene* HelloWorld::createScene()
{
    auto scene = HelloWorld::create();
    return scene;
}
```

(3) --- Edit HelloWorldScene.cpp to make init() call Scene::init():

```
bool HelloWorld::init()
{
    if ( !Scene::init() )
    {
        return false;
    }
..............
}
```

Note that this code works fine in v3.6 or older.

**Comment by ricardoquesada** [comment]
looks like a compiler bug... not a cocos2d bug.
But to be sure, you should post the whole project... or try to compile it on Mac, or on Android.
If it works with other compilers, then it is a VS bug.

**Comment by redminion** [comment]
Okay Ricardo, I have created a new repository with a brand new v3.7 project (and my changes),  you can find it here: https://github.com/redminion/cc37final_bugTest

I don't have a Mac to try it on,  but I can try it on Android later today.

**Comment by ricardoquesada** [comment]
Thanks. Yes, please, test it on Android. If it crashes on Android I'll take a look at your test project. Thanks.

**Comment by pandamicro** [comment]
It should have been fixed in https://github.com/cocos2d/cocos2d-x/issues/12906
@yangws Please confirm

**Comment by ricardoquesada** [comment]
good finding. yes. closing it.

**Comment by pandamicro** [comment]
As it's a new issue caused by v3.7, we decided to migrate the fix into v3.7.1

**Comment by ricardoquesada** [comment]
makes sense.
